server/server.js

```diff
@@ -54,8 +54,19 @@ app.use(methodOverride());
 // LoopBack REST interface
 app.use(app.get('restApiRoot'), loopback.rest());
 
-// API explorer
-require('loopback-component-explorer')(app);
+// To prevent the snyk warning about swagger-ui@2.x, the explorer component
+// is moved from dependencies to devDependencies.
+// Here we try to mount the explorer component in case it's not installed.
+
+try {
+  require('loopback-component-explorer')(app);
+} catch (e) {
+  console.log(
```

Comment:
Please check if the error is caused by the missing explorer package or something else. We have been bitten in the past by code that was hiding legitimate errors and masking them as "module was not found".

You can use the code loading connector modules for inspiration.

https://github.com/strongloop/loopback-datasource-juggler/blob/261dd1c865e8b197882a161b909b90b0e6c3c2a3/lib/datasource.js#L262-L271
```js
      const notFound = e.code === 'MODULE_NOT_FOUND' &&
        e.message && e.message.indexOf(names[m]) > 0;


      if (notFound) {
        debug('Module %s not found, will try another candidate.', names[m]);
        continue;
      }


      debug('Cannot load connector %s: %s', names[m], e.stack || e);
      throw e;
```

Reply:
I would also not use `console.log()`, we don't want to spam users of `lb` and `apic` CLI tools. Can we use `debug()` instead?

Reply:
Thanks @bajtos yeah debug is better. And I added the message check for error to destinguish the not found error and others.